gitops/environment.go

```diff
@@ -0,0 +1,92 @@
+package gitops
+
+import (
+	"github.com/google/uuid"
+	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	"github.com/redhat-appstudio/integration-service/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CopiedEnvironment struct {
+	applicationapiv1alpha1.Environment
+}
+
+// AsEnvironment casts the IntegrationPipelineRun to PipelineRun, so it can be used in the Kubernetes client.
+func (r *CopiedEnvironment) AsEnvironment() *applicationapiv1alpha1.Environment {
+	return &r.Environment
+}
+
+// NewCopyOfExistingEnvironment gets the existing environment from current namespace and makes copy of it
+// new name is generated consisting of existing environment name and integrationTestScenario name
+// targetNamespace gets name of integrationTestScenario and uuid
+func NewCopyOfExistingEnvironment(existingEnvironment *applicationapiv1alpha1.Environment, namespace string, integrationTestScenario *v1alpha1.IntegrationTestScenario) *CopiedEnvironment {
+	id := uuid.New()
+	existingApiURL := existingEnvironment.Spec.UnstableConfigurationFields.KubernetesClusterCredentials.APIURL
+	existingClusterCreds := existingEnvironment.Spec.UnstableConfigurationFields.KubernetesClusterCredentials.ClusterCredentialsSecret
+
+	copiedEnvConfiguration := applicationapiv1alpha1.EnvironmentConfiguration{}
+	copiedEnvConfiguration = *existingEnvironment.Spec.Configuration.DeepCopy()
+	if existingEnvironment.Spec.Configuration.Env == nil {
+		copiedEnvConfiguration.Env = integrationTestScenario.Spec.Environment.Configuration.Env
+	} else if len(integrationTestScenario.Spec.Environment.Configuration.Env) != 0 {
+		for intEnvVars := range integrationTestScenario.Spec.Environment.Configuration.Env {
+			// if existing environment does not contain EnvVars, copy ones from IntegrationTestScenario
```

Comment:
I think this comment needs to go above [line 29](https://github.com/redhat-appstudio/integration-service/pull/74/files#diff-d71bb8ba7ed94d9af7d7d17e63cd8c470d40de674d3a3edb9a74121c0c2e9e08R29).

Reply:
Correct, thanks. Updated!